### PR TITLE
feat: add JsonLd

### LINKED ISSUE #649: Add proper structured data (json-ld schema) to optimise for rich-text results
## References

- https://search.google.com/test/rich-results
- https://search.google.com/structured-data/testing-tool

### PR DESCRIPTION
**Pull request by renomureza**
Closes  #649 

## Description

We implement Schema Webpage on all pages, except the home page using the Website.

Webpage combined with JsonLd Breadcrumb.

Here's a list of the Schema Markup we added:

- Organization
- Website
- Webpage

https://user-images.githubusercontent.com/49445216/129440007-067ecb7f-9bb9-427d-b0d8-a70e5d5ae829.mp4

### CODE REVIEW

**Comment by zainfathoni** [comment]
When I tried testing the deploy preview using Google's [Rich Results test](https://search.google.com/test/rich-results), it's still failing due to a response header.

https://search.google.com/test/rich-results?view=fetch-info&id=2PiMx8JOys2wo4UR_SXcSw

Screenshot 2021-08-14 at 19 17 25

I think it's related to our response headers, please try to fix it, @renomureza. Thanks!

Do you have any idea on how to solve this issue, @resir014? 😁

**Comment by mazipan** [comment]
Please list down on the description, what JSON-LD do you add on this PR.

File: components/jsonld-builder.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Head from "next/head";
```
Comment (mazipan) [inline components/jsonld-builder.tsx:1]: Should we grouping this file? Since we have 3 files here.

Maybe `components/json-ld/builder.tsx`

**Comment by renomureza** [comment]
Sorry mas @mazipan , I accidentally pressed the Update Branch button. I'm learning to contribute, please guide.

I'm checking Headers issue. when we remove `interest-cohort=()` the problem showing in console is gone. We can learn about this header here: https://web.dev/floc/

image

**Comment by zainfathoni** [comment]
> I'm checking Headers issue. when we remove `interest-cohort=()` the problem showing in the console is gone. We can learn about this header here: https://web.dev/floc/

I don't think we can simply remove that `interest-cohort=()` header. It's there for security reasons.
@resir014, do you have a better suggestion to solve this issue without imposing security vulnerability on our website?

File: components/jsonld-builder.tsx (hunk 1 above)
Comment (renomureza) [reply components/jsonld-builder.tsx]: Mas @mazipan, we've grouped that file and updated the PR description.

**Comment by renomureza** [comment]
> > I'm checking Headers issue. when we remove `interest-cohort=()` the problem showing in the console is gone. We can learn about this header here: https://web.dev/floc/
> 
> I don't think we can simply remove that `interest-cohort=()` header. It's there for security reasons.
> @resir014, do you have a better suggestion to solve this issue without imposing security vulnerability on our website?

Mas @zainfathoni, this error does not appear in Firefox.

Firefox:

image

Chrome:

**Comment by zainfathoni** [comment]
> Firefox:
> 
> image

@resir014 I think there's another problem that I noticed here. Apparently, Google Analytics is blocked by our Content Security Policy. Could you please address this issue as well? 😁 

> Chrome:
> 
> image

That error means that Chrome doesn't recognize `interest-cohort` feature in the `Permissions-Policy` header. Is that a known issue, @resir014?

**Comment by resir014** [comment]
> I don't think we can simply remove that interest-cohort=() header. It's there for security reasons.
@resir014, do you have a better suggestion to solve this issue without imposing security vulnerability on our website?

It's less security, and more privacy: https://amifloced.org/

Besides, this only exists in Chrome and not other browsers.

**Comment by resir014** [comment]
For the Google Analytics CSP rule, I fixed it in #735. Please check.

**Review by mazipan — APPROVED** [review]
Looks good to me

**Pull request merged**